src/main/kotlin/org/jitsi/nlj/transform/node/Node.kt

```diff
@@ -170,42 +171,45 @@ sealed class StatsKeepingNode(name: String): Node(name) {
     }
 
     override fun getNodeStats(): NodeStatsBlock {
-        return NodeStatsBlock("Node $name ${hashCode()}").apply {
-            addStat("numInputPackets: $numInputPackets")
-            addStat("numOutputPackets: $numOutputPackets")
-            addStat("numDiscardedPackets: $numDiscardedPackets")
-            addStat("total time spent: ${Duration.ofNanos(totalProcessingDuration).toMillis()} ms")
-            addStat("average time spent per packet: ${Duration.ofNanos(totalProcessingDuration / Math.max(numInputPackets, 1)).toNanos()} ns")
-            addStat("$numInputBytes bytes over ${Duration.ofNanos(lastPacketTime - firstPacketTime).toMillis()} ms")
-            addStat("throughput: ${getMbps(numInputBytes, Duration.ofNanos(lastPacketTime - firstPacketTime))} mbps")
-            addStat("processing throughput: ${getMbps(numInputBytes, Duration.ofNanos(totalProcessingDuration))} mbps")
-        }
+        val block = NodeStatsBlock("Node $name ${hashCode()}")
+        stats.appendTo(block)
+        block.addStat(
+            "${stats.numInputBytes} bytes over ${Duration.ofNanos(lastPacketTime - firstPacketTime).toMillis()} ms")
+        block.addStat(
+            "throughput: ${getMbps(stats.numInputBytes, Duration.ofNanos(lastPacketTime - firstPacketTime))} mbps")
+
+        return block
```

Comment:
i still think the `apply` scheme looks nicer:
```
return NodeStatsBlock("Node $name ${hashCode()}").apply {
    stats.appendTo(this)
    addStat("${stats.numInputBytes} bytes over ${Duration.ofNanos(lastPacketTime - firstPacketTime).toMillis()} ms")
    addStat("throughput: ${getMbps(stats.numInputBytes, Duration.ofNanos(lastPacketTime - firstPacketTime))} mbps")
}
```

Reply:
I actually tried that, but `NodeBlockStats.stats` shadows `stats`. I'm trying to find how to specify that I want `StatsKeepingNode.stats`, but I doubt there will be a nice way to do it (it will have to go in the string template too).

Reply:
```kotlin
    override fun getNodeStats() = NodeStatsBlock("Node $name ${hashCode()}").apply {
        this@StatsKeepingNode.stats.appendTo(this)
        addStat("${this@StatsKeepingNode.stats.numInputBytes} bytes over ${Duration.ofNanos(lastPacketTime - firstPacketTime).toMillis()} ms")
        addStat(
            "throughput: ${getMbps(this@StatsKeepingNode.stats.numInputBytes, Duration.ofNanos(lastPacketTime - firstPacketTime))} mbps")
    }
```